content/docs/5.1/concepts/explore-tikv-features/cas.md

```diff
@@ -0,0 +1,119 @@
+---
+title: CAS on RawKV
+description: Compare And Swap
+menu:
+    "5.1":
+        parent: Features
+        weight: 4
+---
+
+This page walks you through a simple demonstration of how to use TiKV's `compare-and-swap`.
+
+In RawKV, `compare-and-swap (CAS)` is an atomic instruction used in multi-threading to achieve synchronization. It compares the `current_value` of a `key` with a given value and, only if they are the same, modifies the contents of that `key` to a new given value. This is done as a single atomic operation. The atomicity guarantees that the new value is calculated based on up-to-date information; if the value had been updated by another thread in the meantime, the write would fail.
+
+## Prerequisites
+
+1. Install TiUP by executing the following command:
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
+```
+
+2. If TiUP is already installed, update the TiUP playground component to the latest version:
+
+```bash
+tiup update --self && tiup update playground
+```
+
+3. Install Rust
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+## Step 1: Create a rust project
+
+Use `cargo` to create a rust project:
+
+```bash
+cargo new hello-tikv
+cd hello-tikv
+```
+
+## Step 2: Add dependency
+
+Append the following content to file `Cargo.toml`.
+
+```toml
+tikv-client = "0.1.0"
+futures = "0.3"
+```
+
+## Step 3: Write test code
+
+Save the following code to file `src/main.rs`.
+
+```rust
+use tikv_client::RawClient;
+use futures::executor::block_on;
+
+fn main() {
+    block_on(raw_cas());
+}
+
+async fn raw_cas() {
+    let client = RawClient::new(vec!["127.0.0.1:2379"])
+        .await.unwrap()
+        // Users must set `with_atomic_for_cas` in `put`, `delete`,
+        // `batch_delete` and `batch_put` to ensure linearizability
+        // of `compare_and_swap`.
+        .with_atomic_for_cas();
+
+    let key = "key".to_owned();
+    let value = "value".to_owned();
+    let new_value = "newValue".to_owned();
+
+    // call put(key, value)
+    client.put(key.clone(), value.clone()).await.unwrap();
+    println!("put key={}, value={}", key, value);
+
+    // call get(key) returns value
+    let res = client.get(key.clone()).await.unwrap();
+    println!("get key={} returnedValue {:?}", key, res);
+
+    // call compare_and_swap(key, value, new_value) returns (value, true)
+    // If the value retrived is equal to `current_value`, `new_value` is written.
+    // A tuple is returned if successful: the previous value and whether the value is swapped
+    let res = client
+        .compare_and_swap(
+            key.clone(),
```

Comment:
could we use reference instead of clone?

Reply:
use `to_owned` instead

Reply:
`Key` implement this trait https://docs.rs/tikv-client/0.1.0/src/tikv_client/kv/key.rs.html#160-162, so and `&Vec<u8>` could be use as argument in `compare_and_swap`